src/main/java/build/buildfarm/worker/CASFileCache.java

```diff
@@ -2647,6 +2664,9 @@ private static void setPermissions(Path path, boolean isExecutable) throws IOExc
     int referenceCount;
     Deadline existsDeadline;
 
+    public static int containedDirectoriesMax = 0;
```

Comment:
This must not be static, you will collide with other instantiations of the CASFileCache: it must be a member on the outer class, non-static.